go/vt/vttablet/tabletserver/connpool/pool.go

```diff
@@ -200,16 +205,18 @@ func (cp *Pool) Get(ctx context.Context, setting *pools.Setting) (*DBConn, error
 		ctx, cancel = context.WithTimeout(ctx, cp.timeout)
 		defer cancel()
 	}
+
 	start := time.Now()
 	r, err := p.Get(ctx, setting)
 	if err != nil {
 		return nil, err
 	}
-	timeElapsed := time.Since(start)
-	if setting == nil {
-		cp.getConnTime.Set("without", timeElapsed.Nanoseconds())
-	} else {
-		cp.getConnTime.Set("with", timeElapsed.Nanoseconds())
+	if cp.getConnTime != nil {
+		if setting == nil {
+			cp.getConnTime.Record(getWithoutS, start)
+		} else {
+			cp.getConnTime.Record(getWithS, start)
```

Comment:
This is just recording the time when the Get function was called. It is not recording the time that it took to finish execution. Is that what we want? If that is the case, then the metric's help should change, because currently, it says `Tracks the amount of time it takes to get a connection`

Reply:
the record function does the diff for us. it is a helper function to the Add in timing stats.

Reply:
```go
// Record is a convenience function that records completion
// timing data based on the provided start time of an event.
func (t *Timings) Record(name string, startTime time.Time) {
	if t.labelCombined {
		name = StatsAllStr
	}
	t.Add(name, time.Since(startTime))
}

```

Reply:
Oh okay 👍